resources/uco_design_document.md

```diff
@@ -181,7 +213,35 @@ By default the UUIDs used in object identifiers should be randomly generated UUI
+
+As discussed in #3 above, there are some limitations/downsides to the common default purpose-specific object property approach for associating instances of `UcoObject` with instances of `UcoInherentCharacterizationThing` that characterize them.
+
+Some of these are:
+
+* Adding any new `UcoInherentCharacterizationThing` classes requires adding new purpose-specific object properties as well as changes to the shape of the `UcoObject` class it applies to.
+  * In some cases (where purpose-specific object properties may not be appropriate) this leads to unnecessarily verbose/complex serializations.
+  * In some cases (where purpose-specific object properties may not be appropriate) this means defining, managing, tracking and processing many unnecessary object properties. Each new object property is a new predicate and therefore a new type of graph edge.
+  * In many cases (where purpose-specific object properties may not be appropriate) this leads to unnecessary and sometimes impractical churn on the UcoObject shapes affecting adopters.
+     * This is particularly true for new `UcoInherentCharacterizationThing` classes added to application subdomain ontologies built on UCO and to very specific usage communities and scenarios.
+         * In many of these cases the new `UcoInherentCharacterizationThing` classes may only be relevant to the application subdomain ontologies or very specific usage communities and scenarios. In these cases modifying the general shape for the `UcoObject` at the UCO level would be inappropriate.
+     * This is also true for defining particular subclasses of existing `UcoInherentCharacterizationThing` classes
+* In some cases (e.g., Identity) (where purpose-specific object properties may not be appropriate) this also makes it more complicated for particular adopters (e.g., application subdomain ontologies or even specific communities or organizations) to scope their adoption to specific `UcoInherentCharacterizationThing` class-based characteristics.
+  * The purpose-specific object property approach requires them to define their own specialized complex shapes (defining property shapes for all of the separate purpose-specific object properties) for the relevant UcoObjects.
+* In some cases (e.g., Location) (where purpose-specific object properties may not be appropriate) this also makes it unnecessarily more complicated for expressing the same basic concept characterization using different mechanisms (e.g., a location expressed utilizing different coordinate systems).
+* In some cases where specific context may not yet be understood and more general property association is necessary (e.g. duck typing), the purpose-specific object properties approach is impractical.
```

Comment:
The statement on impracticality is unsubstantiated.  In my opinion, this needs a worked example.